src/snowflake/connector/s3_storage_client.py

```diff
@@ -197,48 +263,82 @@ def _send_request_with_authentication_and_retry(
         self,
         url: str,
         verb: str,
-        resources: str,
         retry_id: Union[int, str],
+        query_parts: Optional[Dict[str, str]] = None,
         x_amz_headers: Optional[Dict[str, str]] = None,
         headers: Optional[Dict[str, str]] = None,
-        content_type: str = "",
-        data: Union[bytes, bytearray, IOBase, None] = None,
+        payload: Union[bytes, bytearray, IOBase, None] = None,
+        unsigned_payload: bool = False,
     ) -> requests.Response:
-        if not x_amz_headers:
+        if x_amz_headers is None:
             x_amz_headers = {}
-        if not headers:
+        if headers is None:
             headers = {}
+        if payload is None:
+            payload = b""
+        if query_parts is None:
+            query_parts = {}
```

Comment:
Here I define some better default values, for example if `x_amz_headers={}` then that would have redefined it.